src/storage.ts

```diff
@@ -52,14 +52,28 @@ export class GlobalStorage {
    * to create a more specific Uri.
    */
   getHistoryUri(...segments: string[]): vscode.Uri {
-    // Example: $USER/Library/Application Support/Code/User/globalStorage/undefined_publisher.vscode-scriptiq/v1/scriptiq_history
+    // Example: /Users/$USER/Library/Application\ Support/Code/User/globalStorage/saucelabs.vscode-scriptiq/v1/scriptiq_history/
     return vscode.Uri.joinPath(
       this.storageUri,
       'scriptiq_history',
       ...segments,
     );
   }
 
+  // Verifies the existence of test IDs by checking their corresponding URIs on the disk.
+  // Returns a filtered array containing only the IDs that have corresponding URIs.
+  verifyTestIDs(ids: string[]): string[] {
+    return ids.filter((id) => {
+      const uri = this.getHistoryUri(id);
+      try {
+        return fs.existsSync(uri.fsPath);
+      } catch (err) {
+        console.error(`Failed to check existence for ${uri.fsPath}: ${err}`);
```

Comment:
"Failed to check existence" — that'd mean that this function failed to perform the file-exists check, not that the file doesn't exist ⚠️ 
```suggestion
        console.error(`Test Record ${uri.fsPath} not found: ${err}`);
```

Reply:
hmm.. If the test record doesn't exist, `fs.existsSync` will return false, not throw error. The catch block is catching unexpected err 🤔 . I can add a log to indicate the file doesn't exist.

Reply:
Update: `fs.existsSync` doesn't throw any error. I just remove the try catch block.